counsel.el

```diff
@@ -5234,6 +5236,19 @@ You may, for example, want to add \"-jN\" for the number of cores
 N in your system."
   :type 'string)
 
+(defcustom counsel-compile-env nil
+  "List of environment variables for compilation to inherit.
+Each element should be a string of the form ENVVARNAME=VALUE.  This
+list is passed to `compilation-environment'."
+  :type '(repeat (string :tag "ENVVARNAME=VALUE")))
+
+(defvar counsel-compile-env-history nil
+  "History for `counsel-compile-env'.")
+
+(defvar counsel-compile-env-pattern
+  "\\(?:^[_[:digit:][:upper:]]+=[/[:alnum:]]*\\)"
```

Comment:
You never answered my question as to why the shy group (`\\(?:...\\)`) is needed.

Reply:
Ah, maybe that was an artefact of writing it in rx.